panos_upgrade_assurance/firewall_proxy.py

```diff
@@ -254,21 +254,16 @@ def is_panorama_connected(self) -> bool:
         if not isinstance(pan_status, str):
             raise exceptions.MalformedResponseException("Response from device is not type of string.")
 
-        pan_status_list = pan_status.split("\n")
-        pan_status_list_length = len(pan_status_list)
-
-        if pan_status_list_length in [3, 7]:
-            for i in range(1, pan_status_list_length, 4):
-                pan_connected = interpret_yes_no((pan_status_list[i].split(":")[1]).strip())
-                if pan_connected:
-                    return True
+        panorama_connection_pattern = r"connected\s*:\s*(yes|no)"
+        match = re.search(panorama_connection_pattern, pan_status, re.IGNORECASE)
```

Comment:
If we match yes|no and the first Panorama is disconnected while the second one is connected, method would incorrectly return False as if not connected. I think we can directly match "yes" to return connected, and in the elif statement we can match for "no" to return False; and in the else statement we can raise exception where we couldn't find a match.